packages/lib/src/messaging/Messaging.ts

```diff
@@ -70,37 +83,71 @@ export enum MessageState {
     FailedToSend,
 }
 
-export function createMessage(
-    to: string,
-    from: string,
-    message: string,
-    getTimestamp: () => number,
-    type: MessageType,
-    signature: string,
-    referenceMessageHash?: string,
-    attachments?: string[],
-    replyDeliveryInstruction?: string,
-): Message {
+export interface SendDependencies {
+    from: Account;
+    to: Account;
+    deliveryServiceEncryptionPubKey: string;
+    keys: ProfileKeys;
+}
+
+export async function buildEnvelop(
+    message: Message,
+    encryptAsymmetric: EncryptAsymmetric,
+    { to, from, deliveryServiceEncryptionPubKey, keys }: SendDependencies,
+): Promise<{ encryptedEnvelop: EncryptionEnvelop; envelop: Envelop }> {
+    if (!to.profile) {
+        throw Error('Contact has no profile');
+    }
+
+    const encryptedMessage = stringify(
+        await encryptAsymmetric(
+            to.profile.publicEncryptionKey,
+            stringify(message),
+        ),
+    );
+
+    const deliveryInformation: DeliveryInformation = {
+        to: to.address,
+        from: from.address,
+    };
+
+    const envelopeMetadata: Omit<EnvelopeMetadata, 'signature'> = {
+        encryptionScheme: 'x25519-chacha20-poly1305',
```

Comment:
maybe it could make sense to create a global constant file to define constants like the encryption scheme or the version at one place

Reply:
#367